framework/test/jest/specs/unit/controller/channels/child_process_channel.spec.js

```diff
@@ -0,0 +1,474 @@
+		actions: {
+			action1: jest.fn(),
+			action2: jest.fn(),
+			action3: jest.fn(),
+		},
+		options: {},
+	};
+	const socketsPath = {
+		root: 'root',
+		sub: 'sub',
+		pub: 'pub',
+		rpc: 'rpc',
+	};
+
+	let childProcessChannel;
+	let spies;
+	beforeEach(() => {
+		childProcessChannel = new ChildProcessChannel(
+			params.moduleAlias,
+			params.events,
+			params.actions,
+			params.options
+		);
+
+		spies = {
+			setupSockets: jest
+				.spyOn(childProcessChannel, 'setupSockets')
+				.mockResolvedValue(),
+		};
+	});
+
+	describe('inheritance', () => {
+		it('should be extended from BaseChannel class', () => {
+			// Assert
+			expect(Object.getPrototypeOf(childProcessChannel)).toBeInstanceOf(
+				BaseChannel
+			);
+		});
+
+		it('should call BaseChannel class constructor with arguments', () => {
+			// Arrange
+			let IsolatedChildProcessChannel = null;
+			let IsolatedBaseChannel = null;
+
+			jest.isolateModules(() => {
+				jest.doMock('../../../../../../src/controller/channels/base_channel');
+				IsolatedChildProcessChannel = require('../../../../../../src/controller/channels/child_process_channel');
+				IsolatedBaseChannel = require('../../../../../../src/controller/channels/base_channel');
+			});
+
+			// Act
+			childProcessChannel = new IsolatedChildProcessChannel(
+				params.moduleAlias,
+				params.events,
+				params.actions,
+				params.options
+			);
+
+			// Assert
+			expect(IsolatedBaseChannel).toHaveBeenCalledWith(
+				params.moduleAlias,
+				params.events,
+				params.actions,
+				params.options
+			);
+		});
+	});
+
+	describe('#constructor', () => {
+		it('should create a local bus based on EventEmitter2', () => {
+			// Assert
+			expect(childProcessChannel.localBus).toBeInstanceOf(EventEmitter2);
+		});
+	});
+
+	describe('#registerToBus', () => {
+		beforeEach(() => {
+			childProcessChannel.registerToBus(socketsPath);
+		});
+
+		it('should connect pubSocket', async () => {
+			// Assert
+			expect(childProcessChannel.pubSocket.connect).toHaveBeenCalledWith(
+				socketsPath.sub
+			);
+		});
+
+		it('should connect subSocket', () => {
+			// Assert
+			expect(childProcessChannel.subSocket.connect).toHaveBeenCalledWith(
+				socketsPath.pub
+			);
+		});
+
+		it('should connect busRpcSocket', () => {
+			// Assert
+			expect(childProcessChannel.busRpcSocket.connect).toHaveBeenCalledWith(
+				socketsPath.rpc
+			);
+		});
+
+		it('should expose "invoke" event on rpcServer and call this.invoke with action', () => {
+			// Assert
+			expect(childProcessChannel.rpcServer.expose).toHaveBeenCalledWith(
+				'invoke',
+				expect.any(Function)
+			);
+		});
+
+		it('should bind the rpcSocket to rpcSocketPath', () => {
+			// Assert
+			expect(childProcessChannel.rpcSocket.bind).toHaveBeenCalledWith(
+				childProcessChannel.rpcSocketPath
+			);
+		});
+
+		it('should call setupSockets', () => {
+			// Assert
+			expect(spies.setupSockets).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe('#subscribe', () => {
```

Comment:
subscribe function should give a warning when event is not registered yet. So please add a todo test for that case. The same goes for publish and once.